node/src/net/light.rs

```diff
@@ -177,11 +177,59 @@ impl LightNetwork {
         }
     }
 
+    async fn store<DHT, K, V>(
+        &self,
+        dht: &Arc<DHT>,
+        node_id: NodeId,
+        value: V,
+    ) -> Result<(), LightError>
+    where
+        DHT: ConcreteDht<K> + DhtContent<K, V>,
+    {
+        let encoded = DHT::encode(value)?;
+        if node_id == self.light_peers.node_id {
+            dht.insert_to_store(encoded)?;
```

Comment:
This is app-specific. 
The node ID matching should be done at a higher level. Maybe think about separating client functions (call to other nodes) and local functions (call to database, cache, ...)

Reply:
Fine because we would have to handle the node ID anyway by i.e. returning an error.